### PR TITLE
Add check if no pods are present

### PR DESCRIPTION
**Pull request by varungup90**

### CODE REVIEW

File: pkg/plugins/gateway/algorithms/least_request.go [hunk 1]
```diff
@@ -77,5 +78,9 @@ func (r leastRequestRouter) Route(ctx context.Context, pods map[string]*v1.Pod)
 		}
 	}
 
+	if len(pods) == 0 || targetPodIP == "" {
```
Comment (Jeffwan) [inline pkg/plugins/gateway/algorithms/least_request.go]: let's add some tests for the Route interface.

Comment (Jeffwan) [reply pkg/plugins/gateway/algorithms/least_request.go]: for the pods list empty case, should we return earlier?

Comment (varungup90) [reply pkg/plugins/gateway/algorithms/least_request.go]: Done.

File: benchmarks/gateway/7b.yaml [hunk 2]
```diff
@@ -1,172 +0,0 @@
-apiVersion: apps/v1
```
Comment (Jeffwan) [inline benchmarks/gateway/7b.yaml]: @varungup90 I see some benchmark/plot changes have been added in the PR. are they expected or rebase issue?

**Pull request merged**